datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxHorizontalScalingService.java

```diff
@@ -99,46 +121,83 @@ public String triggerScalingFlow(SdxCluster sdxCluster, StackScaleV4Request scal
+        Map<String, Set<InstanceMetaDataV4Response>> instanceMap = getInstanceMap(stack, target, gatewayInstanceName);
+        if (!instanceMap.containsKey(target)) {
             LOGGER.warn("The requested hostgroup name not found!");
-            return String.format("The requested hostgroup name not found! The requested host group name is %s", target);
+            return String.format("The requested hostgroup name not found! The requested hostgroup name is %s", target);
+        }
+        if (!instanceMap.containsKey(gatewayInstanceName)) {
+            LOGGER.error("Gateway node is not found in the Stack Instance Map. StackId: {}", stack.getId());
+            List<String> instanceGroupList = stack.getInstanceGroups().stream().map(InstanceGroupV4Base::getName).toList();
+            return String.format("The Gateway instance not found, found instances: %s", instanceGroupList);
+        }
+        List<InstanceMetaDataV4Response> gatewayMetaData = getGatewayMetaData(gatewayInstanceName, instanceMap);
+        if (!gatewayMetaData.isEmpty()) {
+            LOGGER.warn("Gateway is not healthy for Data lake horizontal scaling");
+            List<InstanceStatus> instanceStatuses = gatewayMetaData.stream().map(InstanceMetaDataV4Response::getInstanceStatus).toList();
+            return String.format("Gateway instances are not healthy therefore the Data lake horizontal scaling is not executable. " +
+                    "Please Repair the Gateway hostgroup and try again. Actaul Gateway statuses: %s", instanceStatuses);
         }
-        DatalakeInstanceGroupScalingDetails instanceGroupName = DatalakeInstanceGroupScalingDetails.valueOf(request.getGroup().toUpperCase(Locale.ROOT));
-        if (instanceGroupName.getMinimumNodeCount() > request.getDesiredCount()) {
+        DatalakeInstanceGroupScalingDetails targetInstanceGroupName = DatalakeInstanceGroupScalingDetails.valueOf(request.getGroup().toUpperCase(Locale.ROOT));
+        if (targetInstanceGroupName.getMinimumNodeCount() > request.getDesiredCount()) {
             LOGGER.warn("Requested nodeCount is less than the minimum nodeCount.");
             return String.format("Requested nodeCount is less than the minimum nodeCount. The minimum node cont for %s is %d",
-                    instanceGroupName.name(), instanceGroupName.getMinimumNodeCount());
+                    targetInstanceGroupName.name(), targetInstanceGroupName.getMinimumNodeCount());
+        }
+        if (isDownscaleBlocked(targetInstanceGroupName) && isDownscale(stack, request)) {
+            return "The storage hostgroup down scale is not supported, because it can cause data lost";
         }
         return "";
     }
+
+    private boolean isDownscaleBlocked(DatalakeInstanceGroupScalingDetails targetInstanceGroupName) {
+        return DatalakeInstanceGroupScalingDetails.STORAGEHG == targetInstanceGroupName
```

Comment:
consistency: next method is using switch/case for similar evaluation. Use `==` *OR* `switch` in both places.